src/utils/locale.ts

```diff
@@ -61,57 +62,44 @@ function createLocaleNumberFormatter(locale: string): Intl.NumberFormat {
   return sanitizeExponentialNumberString(numberString, (nonExpoNumString: string): string => {
-    if (nonExpoNumString) {
-      const groupSeparator = getGroupSeparator(locale);
-      const decimalSeparator = getDecimalSeparator(locale);
-      const minusSign = getMinusSign(locale);
-
-      const splitNumberString = nonExpoNumString.split("");
-      const decimalIndex = splitNumberString.lastIndexOf(decimalSeparator);
-
-      const delocalizedNumberString = splitNumberString
-        .map((value, index) => {
-          if (value === groupSeparator || (value === decimalSeparator && index !== decimalIndex)) {
-            return "";
-          }
-          return value;
-        })
-        .reduce((string, part) => string + part)
-        .replace(decimalSeparator, ".")
-        .replace(minusSign, "-")
-        .replace(/[^0-9\-\.]/g, ""); // remove everything except numbers, "-", and "."
+    const allDecimalsExceptLast = new RegExp(`[.](?=.*[.])`, "g");
+    const everythingExceptNumbersDecimalsAndMinusSigns = new RegExp("[^0-9-.]", "g");
+    const delocalizedNumberString = nonExpoNumString
+      .replace(getMinusSign(locale), "-")
+      .replace(getGroupSeparator(locale), "")
+      .replace(getDecimalSeparator(locale), ".")
+      .replace(allDecimalsExceptLast, "")
+      .replace(everythingExceptNumbersDecimalsAndMinusSigns, "");
 
-      return isNaN(Number(delocalizedNumberString)) ? nonExpoNumString : delocalizedNumberString;
-    }
-    return nonExpoNumString;
+    return isValidNumber(delocalizedNumberString) ? delocalizedNumberString : nonExpoNumString;
   });
 }
 
 export function getGroupSeparator(locale: string): string {
   const formatter = createLocaleNumberFormatter(locale);
-  const parts = formatter.formatToParts(1234567.8);
-  const value = parts.find((part) => part.type === "group").value;
-  return value.trim().length === 0 ? " " : value;
+  const parts = formatter.formatToParts(1234567);
+  const value = parts.find((part) => part.type === "group")?.value;
+  return value ? value.trim() : "";
 }
 
 export function getDecimalSeparator(locale: string): string {
   const formatter = createLocaleNumberFormatter(locale);
-  const parts = formatter.formatToParts(1234567.8);
-  const value = parts.find((part) => part.type === "decimal").value;
-  return value.trim().length === 0 ? " " : value;
+  const parts = formatter.formatToParts(1.1);
+  const value = parts.find((part) => part.type === "decimal")?.value;
+  return value ? value.trim() : "";
 }
 
 export function getMinusSign(locale: string): string {
   const formatter = createLocaleNumberFormatter(locale);
-  const parts = formatter.formatToParts(-1234567.8);
-  const value = parts.find((part) => part.type === "minusSign").value;
-  return value.trim().length === 0 ? "-" : value;
+  const parts = formatter.formatToParts(-9);
+  const value = parts.find((part) => part.type === "minusSign")?.value;
```

Comment:
Is it likely that the decimal and minus sign (maybe group too) parts won't be found? Seems like this wouldn't happen, but maybe I'm missing something.

Reply:
I can't think of a time where the part couldn't be found with the way the functions are now. However, the three functions are a bit redundant, so I'm combining them to

```js
export function getNumberPart(type: string, locale: string): string {
  const formatter = createLocaleNumberFormatter(locale);
  const parts = formatter.formatToParts(-1234567.89);
  return parts.find((part) => part.type === type)?.value || "";
}
```
in which case it could be `undefined` if someone passes an invalid part type. Also, it actually shouldn't be trimmed because some locales group part is `" "`.

Reply:
Sorry I just confused myself a bit. Here is an example of why I did that: https://codepen.io/benesri/pen/KKZxLEK?editors=0012

Reply:
Okay, I figured out why the `return value.trim().length === 0 ? " " : value;` code existed. There are whitespace group characters that won't render correctly. So that is swapping the non-renderable white space with a renderable one. None of the locales have white space for the decimal, so that must have just been due to copy/paste. I added a comment so we don't forget again. cc @eriklharper